### PR TITLE
[Enhancement] support load struct from parquet

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #

1. support load struct from parquet
2. support map/struct on aggreagte key model

```
MySQL td> select * from s2;
+---------+------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
| col_int | col_string | col_struct                                                                                                                                               |
+---------+------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
| 2       | 2-test     | {"c_int":2,"c_float":2.1,"c_double":2.2,"c_char":"char2","c_varchar":"varchar2","c_date":"2020-02-02","c_timestamp":"2020-02-02 20:00:00","c_boolean":1} |
| 1       | 4-test     | {"c_int":4,"c_float":4.1,"c_double":4.2,"c_char":"char4","c_varchar":"varchar4","c_date":"2020-04-04","c_timestamp":"2020-04-04 20:00:00","c_boolean":1} |
| 3       | 3-test     | {"c_int":3,"c_float":3.1,"c_double":3.2,"c_char":"char3","c_varchar":"varchar3","c_date":"2020-03-03","c_timestamp":"2020-03-03 20:00:00","c_boolean":1} |
+---------+------------+----------------------------------------------------------------------------------------------------------------------------------------------------------+
```

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/ColumnDef.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/AggregateType.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/ColumnDef.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/AggregateType.java|1|1|100.00%|[]|

File: be/src/storage/column_aggregate_func.cpp [hunk 1]
```diff
@@ -193,6 +195,49 @@ class ReplaceAggregator<ArrayColumn, ArrayState> final : public ValueColumnAggre
...
+        auto* col = down_cast<MapColumn*>(agg);
+        DCHECK_NOTNULL(this->data().column);
+        col->append(*this->data().column, this->data().row, 1);
+    }
+
+    bool need_deep_copy() const override { return true; }
+};
+
+template <>
+class ReplaceAggregator<StructColumn, ColumnRefState> final
```
Comment (stdpain) [inline be/src/storage/column_aggregate_func.cpp:220]: duplicate implements with Map/ArrayColumn?

Comment (Seaven) [reply be/src/storage/column_aggregate_func.cpp:220]: yeah, almost the same

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/ColumnDef.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/AggregateType.java|1|1|100.00%|[]|

**Review by stdpain — APPROVED** [review]

**Review by liuyehcf — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/ColumnDef.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/AggregateType.java|1|1|100.00%|[]|

**Pull request merged**